mllib/src/main/scala/org/apache/spark/mllib/classification/NaiveBayes.scala

```diff
@@ -83,9 +86,13 @@ class NaiveBayesModel private[spark] (
   }
 
   override def predict(testData: RDD[Vector]): RDD[Double] = {
-    val bcModel = testData.context.broadcast(this)
+    bcModel match {
```

Comment:
That makes sense. I initially started with just this Jira and once the PR was ready, I realized the rest of them had similar fixes! Let me put something together for this one and if it looks good, i will reproduce for the other PRs. I will update this PR shortly. Thx!